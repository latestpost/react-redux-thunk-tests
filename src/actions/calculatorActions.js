export const addInput = (input) => {
    return {
        type: 'ADD_INPUT',
        input: input 
    }
}

export const multiplyInput = (input) => {
    return {
        type: 'MULTIPLY_INPUT',
        input: input 
    }
}

export const fetchSomethingRequest =  () => {
  return {
    type: 'FETCH_SOMETHING_REQUEST'
  }
}

export const fetchSomethingSuccess = (body) => {
  return {
    type: 'FETCH_SOMETHING_SUCCESS',
    body: body
  }
}

export const fetchSomethingFailure = (err) => {
  return {
    type: 'FETCH_SOMETHING_FAILURE',
    err
  }
}

export const fetchSomething = () => {
  return function (dispatch) {
    dispatch(fetchSomethingRequest())
    return fetch('http://1/test', {
    method: 'GET'
  })
  .then(function(result) {
    if (result.status === 200) {
      dispatch(fetchSomethingSuccess())
      return 'result'
    }
    dispatch(fetchSomethingFailure())
    return 'failed'
  }, function(error) {
    return 'error'
  })
  }
}