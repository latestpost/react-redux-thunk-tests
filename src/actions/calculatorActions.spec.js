import { addInput, multiplyInput, fetchSomething } from './calculatorActions'

import configureMockStore from 'redux-mock-store'
import thunk from 'redux-thunk'
import fetchMock from 'fetch-mock'

const middlewares = [thunk]
const mockStore = configureMockStore(middlewares)

describe('>>>A C T I O N --- Test calculatorActions', () => {
    it('+++ actionCreator addInputs', () => {
        const add = addInput(50)
        expect(add).toEqual({ type: "ADD_INPUT", input: 50 })
        expect(add).toMatchSnapshot();
    });
    it('+++ actionCreator multiplyInputs', () => {
        const add = multiplyInput(31)
        expect(add).toMatchSnapshot();
    });
});

describe('Test thunk action creator', () => {
    afterEach(() => {
        fetchMock.reset()
        fetchMock.restore()
    })
    it('expected actions should be dispatched on failed request', () => {
        const store = mockStore({})
        const expectedActions = [
            'FETCH_SOMETHING_REQUEST',
            'FETCH_SOMETHING_FAILURE'
        ]
        fetchMock.get('*', 404)

        return store.dispatch(fetchSomething())
            .then(() => {
                const actualActions = store.getActions().map(action => action.type)
                expect(actualActions).toEqual(expectedActions)
            })
    })

    it('expected actions should be dispatched on successful request', () => {
        const store = mockStore({})
        const expectedActions = [
            'FETCH_SOMETHING_REQUEST',
            'FETCH_SOMETHING_SUCCESS'
        ]

        fetchMock.get('*', 200)
        return store.dispatch(fetchSomething())
            .then(() => {
                const actualActions = store.getActions().map(action => action.type)
                expect(actualActions).toEqual(expectedActions)
            })
    })
})