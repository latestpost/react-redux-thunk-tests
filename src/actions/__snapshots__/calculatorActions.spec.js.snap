// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>>A C T I O N --- Test calculatorActions +++ actionCreator addInputs 1`] = `
Object {
  "input": 50,
  "type": "ADD_INPUT",
}
`;

exports[`>>>A C T I O N --- Test calculatorActions +++ actionCreator multiplyInputs 1`] = `
Object {
  "input": 31,
  "type": "MULTIPLY_INPUT",
}
`;
